packages/flutter_rearch/lib/src/widgets/consumer.dart

```diff
@@ -120,6 +120,11 @@ class _WidgetSideEffectApiProxyImpl implements WidgetSideEffectApi {
   @override
   void unregisterDispose(SideEffectApiCallback callback) =>
       manager.disposeListeners.remove(callback);
+
+  @override
+  void runTransaction(void Function() sideEffectTransaction) =>
+      CapsuleContainerProvider.containerOf(manager)
+          .runTransaction(sideEffectTransaction);
```

Comment:
Add some tests to make sure this won't error out from trying to access stuff from build context at possibly invalid points of the widget life cycle

Reply:
From the Flutter docs:

```dart
  /// It is also possible to call this method from interaction event handlers
  /// (e.g. gesture callbacks) or timers, to obtain a value once, as long as
  /// that value is not cached and/or reused later.
```

So this should be fine. still need to test it though.